src/Parallel/ArrayCollection/SpawnInitializeElementsInCollection.hpp

```diff
@@ -24,7 +24,8 @@ struct SpawnInitializeElementsInCollection {
                     Parallel::GlobalCache<Metavariables>& cache,
                     const ArrayIndex& /*array_index*/,
                     const double /*unused_but_we_needed_to_reduce_something*/) {
-    auto my_proxy = Parallel::get_parallel_component<ParallelComponent>(cache);
+    auto my_proxy = Parallel::get_parallel_component<ParallelComponent>(
+        cache)[Parallel::my_node<size_t>(cache)];
```

Comment:
Just for my understanding, this used to do a broadcast to the whole collection for every element on this node? But that didn't matter because you only ran on a single node? And now you only send the messages to the local node?

Reply:
Right, and the issue was that it would try to run an element that wasn't on its node and would then fail with "I can't find this element!"

Reply:
Thanks for the explanation!